pkg/catalog/egress.go

```diff
@@ -87,6 +88,7 @@ func (mc *MeshCatalog) GetEgressTrafficPolicy(serviceIdentity identity.ServiceId
 	if err != nil {
 		log.Error().Err(err).Str(errcode.Kind, errcode.ErrDedupEgressTrafficMatches.String()).
 			Msgf("Error deduplicating egress traffic matches for service identity %s", serviceIdentity)
+		metricsstore.DefaultMetricsStore.ErrcodeCounter.WithLabelValues(errcode.ErrDedupEgressTrafficMatches.String(), errcode.Egress).Inc()
```

Comment:
Adding an extra log line for each error log just to generate the metric doesn't seem idiomatic. 

I suggest 2 options to solve this:
1. Create a generic utility in `pkg/errcode`:
 ```
getErrCodeWithMetric(e errcode.ErrCode) string {
    metricsstore.DefaultMetricsStore.ErrcodeCounter.WithLabelValues(e.String())
return e.String()
}
...
func someFunc() {
    if err != nil {
        log.Error().Str(errcode.Kind, getErrCodeWithMetric(errcode.SomeError).Msg(...)
    }
}
```

2. Do the magic of capturing error code metrics in the logger's hook

Reply:
Great suggestion! I will try the first option. 

I explored using a hook, but it would have required parsing the log event's buffer to obtain the errcode.  The performance cost of this work made the option less appealing.